File: C11/백준/Gold/13172. Σ/Σ.c
#include <stdio.h>

#define MOD 1000000007

long long mod_pow(long long base, long long exp) {
    long long result = 1;
    while (exp > 0) {
        if (exp & 1) result = (result * base) % MOD;
        base = (base * base) % MOD;
        exp >>= 1;
    }
    return result;
}

int main() {
    int M;
    scanf("%d", &M);
    long long result = 0;
    for (int i = 0; i < M; i++) {
        long long N, S;
        scanf("%lld %lld", &N, &S);
        long long invN = mod_pow(N, MOD - 2);
        result = (result + (S * invN) % MOD) % MOD;
    }
    printf("%lld\n", result);
    return 0;
}

File: Java/백준/Gold/13172. Σ/Σ.java
import java.io.*;
import java.util.*;

public class Main {
    static final long MOD = 1000000007;

    static long modPow(long base, long exp) {
        long result = 1;
        while (exp > 0) {
            if ((exp & 1) == 1) result = (result * base) % MOD;
            base = (base * base) % MOD;
            exp >>= 1;
        }
        return result;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int M = Integer.parseInt(br.readLine());
        long result = 0;
        for (int i = 0; i < M; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            long N = Long.parseLong(st.nextToken());
            long S = Long.parseLong(st.nextToken());
            long invN = modPow(N, MOD - 2);
            result = (result + (S * invN) % MOD) % MOD;
        }
        System.out.println(result);
    }
}